debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresTaskContext.java

```diff
@@ -66,10 +66,6 @@ protected PostgresConnectorConfig config() {
 
     protected void refreshSchema(PostgresConnection connection, boolean printReplicaIdentityInfo) throws SQLException {
         schema.refresh(connection, printReplicaIdentityInfo);
-        // Open transaction unnecessary during task execution
```

Comment:
Is this strictly related to the change at hand?

Reply:
Yes. We need to hold a transaction open so that we do not lose the view of data that the snapshot uses so this block must not commit during catch up streaming. I figured passing the `OffsetContext` here would be more confusing than helpful and would be duplicated by `PostgresStreamingChangeEventSource#isInPreSnapshotCatchUpStreaming`  so I moved the logic around when to commit up a level into the `PostgresStreamingChangeEventSource`.

Reply:
Ok, thanks.